File: emb/core_task/src/types/state.cpp
#include "state.h"

#include <tuple>

namespace types {
bool operator==(const State& lhs, const State& rhs) {
  return std::tuple(lhs.header, lhs.joint_1_position_rad, lhs.joint_1_velocity_rps, lhs.joint_2_position_rad,
                    lhs.joint_2_velocity_rps) == std::tuple(rhs.header, rhs.joint_1_position_rad,
                                                            rhs.joint_1_velocity_rps, rhs.joint_2_position_rad,
                                                            rhs.joint_2_velocity_rps);
}
}  // namespace types

File: src/motor_platform/motor_platform/types/state.py
from typing import NamedTuple

from motor_platform.types.header import Header


class State(NamedTuple):
    header: Header
    joint_1_position_rad: float
    joint_1_velocity_rps: float
    joint_2_position_rad: float
    joint_2_velocity_rps: float

    @staticmethod
    def decode(json_value):
        return State(
            header=Header.decode(json_value["header"]),
            joint_1_position_rad=json_value["joint_1_position_rad"],
            joint_1_velocity_rps=json_value["joint_1_velocity_rps"],
            joint_2_position_rad=json_value["joint_2_position_rad"],
            joint_2_velocity_rps=json_value["joint_2_velocity_rps"],
        )
